snuba/split.py

```diff
@@ -92,7 +93,9 @@ def time_split(dataset, request: Request, *args, **kwargs):
             # evaluation, so we need to copy the body to ensure that the query
             # has not been modified in between this call and the next loop
             # iteration, if needed.
-            result, status = query_func(dataset, copy.deepcopy(request), *args, **kwargs)
+            query_result = query_func(dataset, copy.deepcopy(request), *args, **kwargs)
+            result = query_result.result
+            status = query_result.status
```

Comment:
Since `QueryResult` is a namedtuple, why not just replace the assignments with:

```
[result, status] = query_func(dataset, copy.deepcopy(request), *args, **kwargs)
```

(Same applies below.)

Reply:
I did it this way because, with the mypy way of defining tuples, their definition looks like a dataclass, instead of declaring the list of fields in sequence.
I can imagine people accidentally reordering the fields in that class not thinking about the impact it has (as a named tuple). Thus accessing the named tuple fields positionally seems dangerous to me.
I suspect, if this way of defining named tuples, becomes the norm, accessing them positionally will become an anti-pattern.

What do you think ?

Reply:
I think that makes sense, as long as the longer-term plan is to move this towards a dataclass.

From an (almost*) purely stylistic perspective, I do think that if we're going to use this as a class, it'd be better to just pass the object around and do attribute accesses rather than assign these to local variables, but I understand if you're trying to minimize the +/- of this change.

* I think you could also make an argument about this making the effect of any mutation side effects less obvious as well, so it's not *purely* stylistic.

Reply:
I did not pass the object around just to make the change easier, I will update it here to pass the object around